x11-wm/awesome/awesome-4.0.ebuild

```diff
@@ -0,0 +1,123 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI="6"
+inherit cmake-utils eutils pax-utils
+
+DESCRIPTION="A dynamic floating and tiling window manager"
+HOMEPAGE="http://awesome.naquadah.org/"
+SRC_URI="https://github.com/awesomeWM/awesome-releases/raw/master/${P}.tar.xz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="dbus doc elibc_FreeBSD gnome"
+REQUIRED_USES="doc? ( amd64 )"
+
+COMMON_DEPEND="
+	>=dev-lang/lua-5.1:0
+	dev-libs/glib:2
+	>=dev-libs/libxdg-basedir-1
+	>=dev-lua/lgi-0.7
+	x11-libs/cairo[xcb]
+	x11-libs/gdk-pixbuf:2
+	>=x11-libs/libxcb-1.6
+	>=x11-libs/pango-1.19.3[introspection]
+	>=x11-libs/startup-notification-0.10_p20110426
+	>=x11-libs/xcb-util-0.3.8
+	x11-libs/xcb-util-cursor
+	>=x11-libs/xcb-util-keysyms-0.3.4
+	>=x11-libs/xcb-util-wm-0.3.8
+	>=x11-libs/xcb-util-xrm-1.0
+	x11-libs/libXcursor
+	x11-libs/libxkbcommon[X]
+	>=x11-libs/libX11-1.3.99.901
+	dbus? ( >=sys-apps/dbus-1 )
+	elibc_FreeBSD? ( dev-libs/libexecinfo )"
+
+# graphicsmagick's 'convert -channel' has no Alpha support, bug #352282
+DEPEND="${COMMON_DEPEND}
+	>=app-text/asciidoc-8.4.5
+	app-text/xmlto
+	dev-util/gperf
+	virtual/pkgconfig
+	media-gfx/imagemagick[png]
+	>=x11-proto/xcb-proto-1.5
+	>=x11-proto/xproto-7.0.15
+	doc? ( amd64? ( dev-lua/ldoc ) )"
+
+RDEPEND="${COMMON_DEPEND}"
+
+DOCS="docs/00-authors.md docs/01-readme.md docs/02-contributing.md"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-4.0-convert-path.patch"  # bug #408025
+	"${FILESDIR}/${PN}-xsession.patch"          # bug #408025
+	"${FILESDIR}/${PN}-4.0-cflag-cleanup.patch" # bug #509658
+)
+
+src_configure() {
+	if use amd64 ; then
+		mycmakeargs=(
+			-DPREFIX="${EPREFIX}"/usr
+			-DSYSCONFDIR="${EPREFIX}"/etc
+			-DWITH_DBUS=$(usex dbus)
+			-DWITH_GENERATE_DOC=$(usex doc)
+		)
+	else
+		mycmakeargs=(
+			-DPREFIX="${EPREFIX}"/usr
+			-DSYSCONFDIR="${EPREFIX}"/etc
+			-DWITH_DBUS=$(usex dbus)
+			-DWITH_GENERATE_DOC=n
+		)
+	fi
+
+	cmake-utils_src_configure
+}
+
+src_install() {
+	cmake-utils_src_install
+
+	pax-mark m "${ED}usr/bin/awesome"
+
+	docompress -x "/usr/share/doc/awesome/doc"
+	einstalldocs
```

Comment:
swap these - not technically a mistake, but it's logically easier to comprehend that you're excluding a directory with stuff, instead of excluding a directory still to be created